src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -0,0 +1,555 @@
+                LinkBlocks(CurrentBasicBlock, afterIf);
+                _currentBasicBlock = null;
+
+                AppendNewBlock(whenFalse);
+
+                AddStatement(new SimpleAssignmentExpression(new FlowCapture(captureId, operation.Syntax,
+                                                                            isInitialization: true, operation.Type,
+                                                                            default(Optional<object>)),
+                                                            operation.IsRef,
+                                                            Visit(operation.WhenFalse),
+                                                            semanticModel: null,
+                                                            operation.WhenFalse.Syntax,
+                                                            operation.Type,
+                                                            default(Optional<object>),
+                                                            isImplicit: true));
+
+                AppendNewBlock(afterIf);
+
+                return new FlowCapture(captureId, operation.Syntax, isInitialization:false, operation.Type, operation.ConstantValue);
+            }
+        }
+
+        private void SpillEvalStack()
+        {
+            for (int i = 0; i < _evalStack.Count; i++)
+            {
+                IOperation operation = _evalStack[i];
+                if (operation.Kind != OperationKind.FlowCapture)
+                {
+                    int captureId = _availableCaptureId++;
+
+                    AddStatement(new SimpleAssignmentExpression(new FlowCapture(captureId, operation.Syntax,
+                                                                                isInitialization: true, operation.Type,
+                                                                                default(Optional<object>)),
+                                                                isRef:false, // PROTOTYPE(dataflow): Is 'false' always the right value?
+                                                                operation,
+                                                                semanticModel: null,
+                                                                operation.Syntax,
+                                                                operation.Type,
+                                                                default(Optional<object>),
+                                                                isImplicit: true));
+
+                    _evalStack[i] = new FlowCapture(captureId, operation.Syntax, isInitialization: false, operation.Type, operation.ConstantValue);
+                }
+            }
+        }
+
+        public override IOperation VisitSimpleAssignment(ISimpleAssignmentOperation operation, object argument)
+        {
+            _evalStack.Push(Visit(operation.Target));
```

Comment:
If the approach I suggested is feasible, then we may not even need an eval stack. We could just hold onto the created `FlowCapture` here and use it a few lines below.

Reply:
> If the approach I suggested is feasible, then we may not even need an eval stack. We could just hold onto the created FlowCapture here and use it a few lines below.

Whether we need a capture depends on whether we have a control flow in the ```operation.Value```. 

---
In reply to: [162175126](https://github.com/dotnet/roslyn/pull/24263#discussion_r162175126) [](ancestors = 162175126)